componentLibraries/defaultLibrary/Signal/Logic/SignalSteadyStateIdentifier.hpp

```diff
@@ -0,0 +1,217 @@
+        }
+
+
+        void initialize()
+        {
+            mWindowId = 0;
+            mWindow.resize(*mpWl/mTimestep, 0);
+
+            mDelayedX = (*mpIn);
+            mDelayedXf = (*mpIn);
+            mDelayedDf = 0;
+
+            if(mWindow.size() <= 2) {
+                stopSimulation("Sliding window is too small compared to time step");
+                return;
+            }
+
+            simulateOneTimestep();
+        }
+
+
+        void simulateOneTimestep()
+        {
+            if(mMethod == 0) {  //Rectangular sliding window
+                mWindow[mWindowId] = *mpIn;
+                mWindowId++;
+                if(mWindowId > mWindow.size()-1) {
+                    mWindowId = 0;
+                }
+
+                double xmax = mWindow[0];
+                double xmin = mWindow[0];
+                for(int i=0; i<mWindow.size(); ++i) {
+                    if(mWindow[i] > xmax) {
+                        xmax = mWindow[i];
+                    }
+                    else if(mWindow[i] < xmin) {
+                        xmin = mWindow[i];
+                    }
+                }
+
+                if(xmax-xmin < *mpTol) {
+                    (*mpOut) = 1;
+                }
+                else {
+                    (*mpOut) = 0;
+                }
+            }
+            else if(mMethod == 1) { //Variance Ratio Test
+                mWindow[mWindowId] = *mpIn;
+                mWindowId++;
+                if(mWindowId > mWindow.size()-1) {
+                    mWindowId = 0;
+                }
+
+                //Randomize window
+                std::vector<double> randWindow = mWindow;
+                for(size_t i=0; i<randWindow.size(); ++i) {
+                    randWindow[i] = randWindow[i] + (*mpSd)*WhiteGaussianNoise::getValue();
+                }
+
+                //Compute average of window
+                double mean = 0;
+                for(size_t i=0; i<randWindow.size(); ++i) {
+                    mean += randWindow[i];
+                }
+                mean /= (double)randWindow.size();
+
+                //Compute first variance
+                double s1 = 0;
+                for(size_t i=0; i<randWindow.size(); ++i) {
+                    s1 += (randWindow[i]-mean)*(randWindow[i]-mean);
+                }
+                s1 /= (double)randWindow.size();
+
+                //Compute second variance
+                //This requires iterating in correct order
+                double s2 = 0;
+                size_t i = mWindowId;
+                size_t j = (i == mWindow.size()-1) ? 0 : mWindowId+1;
+                for(size_t k=0; k<mWindow.size()-1; ++k) {
+                    s2 += (randWindow[j]-randWindow[i])*(randWindow[j]-randWindow[i]);
+                    i = (i == mWindow.size()-1) ? 0 : i+1;
+                    j = (i == mWindow.size()-1) ? 0 : i+1;
+                }
+                s2 /= (double)randWindow.size();
```

Comment:
here there is no -1 as was the case in the gui equivalent code

Reply:
I added it, according to the references.